### PR TITLE
Add a guideline provider checklist 

### PR DESCRIPTION
**Pull request by stevendborrelli**
### Description of your changes

Add a checklist of recommendations for provider writers. This file is being moved from core crossplane (see https://github.com/crossplane/crossplane/pull/2744)

To fix the failing e2e testing I have the following PR: https://github.com/crossplane/provider-template/pull/49

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: PROVIDER_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+## Repository
+
+It is recommended to use the
+[crossplane-contrib](https://github.com/orgs/crossplane-contrib) Github
+Organization. Your project should be named `provider-<name>`. Members of the
+Crossplane community are happy to create a repository and configure access.
+Please open an issue in the Crossplane [org](https://github.com/crossplane/org)
+repository or reach out to the Crossplane
+[#dev](https://crossplane.slack.com/archives/CEF5N8X08) channel.
```
Comment (negz) [inline PROVIDER_CHECKLIST.md:28]: We might want to relax this a little. Under the new charter (https://github.com/crossplane/crossplane/pull/3045) we don't really _recommend_ folks use crossplane-contrib more than any other GitHub org. Instead we offer it as a place for providers that would like a neutral home that is under Crossplane's project governance.

**Review by negz — APPROVED** [review]
Thanks @stevendborrelli, this looks good to me!

Under the new charter it's important that we're clear most of what this document covers (e.g. licensing, CI/CD, etc) are sort of the "golden path" but not actually required. I'm pretty sure this is already true - most of the wording in the document talks about what is "usually involved" or "recommended".

**Comment by stevendborrelli** [comment]
PR https://github.com/crossplane/provider-template/pull/49 addresses the e2e failing issue.

File: PROVIDER_CHECKLIST.md (hunk 1 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md]: I've updated the wording. Please let me know what you think

File: cluster/local/integration_tests.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
```
Comment (jbw976) [inline cluster/local/integration_tests.sh:1]: @stevendborrelli can we remove this integration tests file from this PR? Looks like it's included in #49 also, so let's remove it from here and keep this PR as only the provider checklist markdown file.

Comment (stevendborrelli) [reply cluster/local/integration_tests.sh]: It's been removed.

**Pull request merged**